src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleControl.cs

```diff
@@ -2565,6 +2565,9 @@ internal static void WriteConsole(ConsoleHandle consoleHandle, ReadOnlySpan<char
             // We need to chop the output string if it is too long.
             int cursor = 0; // This records the chopping position in output string
             const int MaxBufferSize = 16383; // this is 64K/4 - 1 to account for possible width of each character.
+            var endOfLine = Environment.NewLine.AsSpan();
+            var endOfLineLength = endOfLine.Length;
+            Span<char> outBufferLine = stackalloc char[(output.Length % MaxBufferSize) + endOfLineLength];
```

Comment:
This change doesn't make sense.
Only value I see if we would allocate const size buffer, with small size (512 chars), and  slice it as needed if newline.

Reply:
> This change doesn't make sense.

Why do you say so?
We would be allocating the same amount of space (as on line 2590), just pre-calculated, and outside the loop.

I'm not sure I fully understand your proposition of using a small buffer that can be sliced.
Can you elaborate on what would we store in said buffer, and how the slicing would come into play?

Reply:
Original code is not ideal but it does not allocate if no newline. It was a compromise.

---
Best practice for stackalloc is (1) allocate as small (<1Kb) buffer as possible (I think we could use 512 char buffer to cover hot scenarios), (2) allocate fixed size so that compiler can optimize (I think newline scenarios is hot and we can always allocate the stack buffer).

So I suggest you think in this direction.

Reply:
Refactored to remove the if-else block in the while loop.

This solution keeps the current implementation, but moves the base case outside the loop.
I think this work for our goal of removing stackalloc from within the loop.  
@iSazonov, let me know what your thoughts are.